bin/block_generator/block_generator.py

```diff
@@ -33,34 +34,56 @@ def __exit__(self, *exit_args):
 
 # Interface for usage of nil_cli
 class NilCLI:
-    def __init__(self, config_path: str):
-        self.config_path = config_path
+    def __init__(self, cli_config_name=None):
+        self.config_path = cli_config_name
 
     def run_cli(self, args: list[str], pattern=None) -> str:
-        full_arg_list = ["nil_cli", "-c", self.config_path] + args
+        full_arg_list = ["nil_cli"]
+        if self.config_path is not None:
+            full_arg_list += ["-c", self.config_path]
+        full_arg_list += args
         logging.info("Launching " + " ".join(full_arg_list))
         p = subprocess.run(full_arg_list, capture_output=True)
-        logging.debug(p.stderr)
         if p.returncode != 0:
             raise RuntimeError(p.stderr)
-        output = p.stderr.decode()
+        output = p.stdout.decode()
+        # result of "config show" command printed to stderr
+        if not output:
+            output = p.stderr.decode()
+        logging.debug(output)
         if pattern is None:
             return output
         a = re.search(pattern, output)
         if a is None:
-            raise RuntimeError(p.stderr)
+            return RuntimeError(p.stderr)
```

Comment:
It must be a typo, usually in Python errors are not supposed to be returned. And I can't find where you check the error later.

Reply:
good catch